Galaxy Proteomics EU Tool YAML Update

jraysajulga opened the pull request:
Added several tools to the YAML file.

bgruening commented:
@jraysajulga this contains a lot more then just proteomics and many tools are already present. But in other files. Can you figure out which tools are really missing in general?

Thanks.

jraysajulga commented:
@bgruening Sure. I will also do some further organization of these tools. One question: How can we add more sections (e.g., GENERAL TEXT TOOLS, FILE AND META TOOLS) to the tool pane?

bgruening commented:
@jraysajulga I can point you to this file in a week, currently its in a private repo ... but in general we would really try to avoid new categories as we try to align these with usegalaxy.eu and usegalaxy.org ... we do have `text processing` tools if this is what you need.

jraysajulga commented:
Okay. That's understandable. We can avoid the new categories then for now.

I just removed tools already present in the tool pane based on this [JSON](https://proteomics.usegalaxy.eu/api/tools?in_panel=true) in that recent commit.

erasche commented:
@jraysajulga no, that's user controlled. Whatever workflows a user favourites show up there.

bgruening commented:
@jraysajulga all tools you add to this repo, will be installed on usegalaxy.eu - this is just one server. *.usegalaxy.eu are just different views, where we hide a different set of tools.
So its a little bit more complicated. All IUC tools are in the tools_iuc.yml file and you don't need to add them here. You really only need to add tools that are missing ... Maybe there are two different problems.

1) Are your tools on `usegalaxy.eu` (all tools can be seen there)?
2) If not, should we show them in `proteomics.usegalaxy.eu`

jraysajulga commented:
@bgruening I removed IUC tools in tools_galaxyp.yaml that were already present in tools_iuc.yaml. However, some IUC tools were not present in the iuc.yaml, so I have some left in the galaxy.yaml. Is that okay? (Example: trinity_gene_to_trans_map)

The majority of tools that I have added are not installed on usegalaxy.eu. We would like to show and use them in proteomics.usegalaxy.eu. Main ones that we want to use for our workflows are as follows:
- regex_find_replace
- custom_pro_db
- gffcompare_to_bed
- translate_bed
- bed_to_protein_map
- concatenate_multiple_datasets
- mz_to_sqlite
- pep_pointer
- peptidegenomiccoordinate

bgruening commented on tools_galaxyp.yaml:
```diff
@@ -1,158 +1,925 @@
+    owner: caleb-easterly
+    tool_panel_section_label: Statistics
+
+  - name: concat_fasta_files
```
This tool is not maintained and also not needed, it just performs a `cat` operation. This can be done by the normal `cat` tools. The reason is that we do need to maintain the tools over time and for this, we would keep the tool number as much as possible.

jraysajulga replied:
I see. Good catch.

bgruening commented on tools_galaxyp.yaml:
```diff
@@ -1,158 +1,925 @@
-    tool_panel_section_label: Proteomics
+    tool_panel_section_label: Sequence Database Tools
+
+  - name: fastq_join
```
This tool is also not needed imho, it can be replaced by https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/flash/flash/1.2.11.3 I think.

bgruening commented on tools_galaxyp.yaml:
```diff
@@ -1,158 +1,925 @@
+    owner: galaxyp
+    tool_panel_section_label: Proteogenomics
+
+  - name: trim_galore
```
that should be already installed on usegalaxy.eu

jraysajulga commented on tools_galaxyp.yaml:
```diff
@@ -1,158 +1,925 @@
 
-  - name: thermo_raw_file_converter
+  - name: peca
+    owner: caleb-easterly
```
I will work towards getting them (PECA, limma_norm, prepare_revigo)  into the galaxyP repository for now if it's okay with @caleb-easterly and we can add them later.

bgruening replied:
limma should/could probably go to IUC, there are already limma tools.

Thanks @caleb-easterly!

bgruening commented on tools_galaxyp.yaml:
```diff
@@ -1,158 +1,925 @@
+    owner: oscar_b
+    tool_panel_section_label: Beta Test Tools
+
+  - name: select_random_lines
```
there is a tool already that does exactly this: https://usegalaxy.eu/root?tool_id=random_lines1

bgruening commented on tools_galaxyp.yaml:
```diff
@@ -1,158 +1,925 @@
+    owner: jjohnson
+    tool_panel_section_label: Beta Test Tools
+
+  - name: spectrast
```
Is that not also available from galaxyp?

bgruening commented:
@jraysajulga I would prefer that the IUC tools are going to the IUC file if this is possible, maybe as separate PR? The reason is, that we do update the IUC file automatically and other not that often.

bgruening commented:
@jraysajulga maybe we do this differently.

Lets do this workflow for workflow... you import a workflow and see if tools missing. If these tools are from IUC, devteam or galaxyp no problem we just add them. If they are from different accounts, lets create an issue with the missing tools and we discuss this ... what do you think about this idea?

jraysajulga commented:
@bgruening  Sounds fair. I recently got a list of workflow tools so this should simplify things quite a bit.

jraysajulga commented:
Okay. Removed all the fluff and only added only the **galaxyp** tools necessary for our workflows.

There was only one tool in our workflows that was not through galaxyp (don't think it's important enough to raise a GH issue). 

**concatenate_multiple_datasets** by _mvdbeek_

You might have an analogous tool available on EU? Personally, I am not familiar with this tool.

bgruening commented on tools_galaxyp.yaml:
```diff
@@ -156,3 +159,32 @@ tools:
     owner: galaxyp
     tool_panel_section_label: Proteomics
+
+  - name: regex_find_replace
```
@jraysajulga do you know offhand why this tool is stricly needed and can not be replaces by https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.1 ... this is my last comment promised :)

jraysajulga replied:
@bgruening concatenate_multiple_datasets can be replaced by another tool so this can be ignored.

As for regex_find_replace, the main issue is that **Replace Text in Entire Line** is unable to take in multiple regex's and replacements. 

In Regex Find and Replace, it is easy to do multiple replacements. We use at least three regular expressions for our workflow. 

Example:
![image](https://user-images.githubusercontent.com/32586466/48634613-95ef1600-e98b-11e8-9f43-5dd4f8955fe0.png)

bgruening replied:
@jraysajulga can you not simply run 3 tools of the classic regex-replace? Or do I miss something?

jraysajulga replied:
@bgruening We could do that if needed.

Is that okay,  @subinamehta?

subinamehta replied:
We could that, but wouldn't the tool be more useful if people have multiple conditions to be fulfilled?

bgruening replied:
@subinamehta yes. But we should rather update the old tool and this is not a critical missing feature or is it. I don't have hard feelings about this. Sorry, I just try to avoid many tools with the same functionality in Galaxy. I would rather try to maintain as less as possible tools over the years to come :)

jraysajulga replied:
@subinamehta We can raise that as a requested functionality in the **Replace Text in entire line** tool. Would that be a GitHub issue?

@bgruening Understandable. I have removed the regex_find_replace tool from yams file.

bgruening commented:
Thanks! :)

bgruening commented:
@subinamehta If I'm not mistaken its this tool: https://github.com/bgruening/galaxytools/blob/master/tools/text_processing/text_processing/find_and_replace.xml any contribution welcome.

jraysajulga commented:
Thanks for installing the tools, @bgruening. Most of them look good except for one: **custom_pro_db**. Can you check this?

bgruening commented:
@jraysajulga did I forget anything here? Is custom_pro_db still a problem?

jraysajulga commented:
@bgruening Thanks for checking up... Looks good now!
